lib/utils/utils.js

```diff
@@ -410,7 +410,7 @@ function getWeiBalanceFromString(balanceString, web3){
     throw new Error(__('Unrecognized balance string "%s"', balanceString));
   }
   if (!match[2]) {
-    return web3.utils.toHex(parseInt(match[1], 10));
+    return match[1];
```

Comment:
This is probably what you're referring to in https://github.com/embark-framework/embark/pull/910/commits/dec098bbebe50c68c6b00683fb3a8001b1f7a813#r221826196

I see where you're coming from, but I'd prefer we don't introduce a breaking change here as part of this commit. Also if we do want to introduce such functionality, we might rather consider introducing an option to alter the current default behaviour to stay backwards compatible.

E.g. sth. like

```
utils.getWeiBalanceFromString(contractGasPrice, web3, 'hex')
```

Maybe something to consider in a separate PR.

However, having that in mind, that means that the tests shouldn't have passed before that change no? Because we're doing a deepEqual here: https://github.com/embark-framework/embark/pull/910/commits/dec098bbebe50c68c6b00683fb3a8001b1f7a813#diff-447140de3e1b662ff6e9201e7352b529L69

Reply:
Oh I just re-read your comment on this:

> prevent it from returning hex values when unitless values are passed.

So this is only happening when unitless values are passed... Hm.. so in that case.. I think I'd still keep the default to be hex, because if we look at this function without the context of this PR:

```
utils.getWeiBalanceFromString('8000000', web3)
```

^ Would you really expect to get `'8000000'` from this call? I think getting a hex value here makes sense as we already have the string number. So in such a scenario, this util function is also a conversion utility. 

The more I think about it, the less I believe this should be changed (or, if we change this, it should be configurable as described above).

Reply:
@PascalPrecht The tests passed before because that spec is testing the conversion of a unit string (here: https://github.com/embark-framework/embark/commit/dec098bbebe50c68c6b00683fb3a8001b1f7a813#diff-08b5381ac801f4b8d8e3a13f70d95fd5R21), so it behaves as it would with the web3 `toWei` function and returns a number string (not a hex string) as shown in the `expectedConfig`. The `getWeiBalanceFromString` function returns a hex string when passed a number string as it currently is written, and a number string when passed a unit string.

Reply:
@PascalPrecht Sounds good - I'm adding a commit now that changes the expected config in the test to contain hex values 👍

Reply:
Please see https://github.com/embark-framework/embark/pull/910#discussion_r221868705

I think the expected config should not contain hex values tho! :D